File: 1/main.py
with open("input.txt") as file:
    data = file.readlines()


def find_first(input_data):
    for x in input_data:
        first = int(x.strip())
        for y in input_data:
            second = int(y.strip())
            result = first + second
            if result == 2020:
                print(f'Result: {first} and {second}')
                multi = first * second
                return multi


def find_second(input_data):
    for x in input_data:
        first = int(x.strip())
        for y in input_data:
            second = int(y.strip())
            for z in input_data:
                third = int(z.strip())
                result = first + second + third
                if result == 2020:
                    print(f'Result: {first}, {second} and {third}')
                    multi = first * second * third
                    return multi


# simple bruteforce solution
first_answer = find_first(data)
print(f'First answer: {first_answer}')
second_answer = find_second(data)
print(f'Second answer: {second_answer}')

File: 1/main.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"path/filepath"
	"strconv"
)

var filename, _ = filepath.Abs("1/input.txt")

func main() {
	data := readLines(filename)
	fmt.Println(firstAnswer(data))
	fmt.Println(secondAnswer(data))

}

func firstAnswer(inputData []string) int {
	for _, x := range inputData {
		var first, _ = strconv.Atoi(x)
		for _, y := range inputData {
			var second, _ = strconv.Atoi(y)
			if first + second == 2020 {
				var result = first * second
				return result
			}
		}
	}
	return 0
}

func secondAnswer(inputData []string) int {
	for _, x := range inputData {
		var first, _ = strconv.Atoi(x)
		for _, y := range inputData {
			var second, _ = strconv.Atoi(y)
			for _, z := range inputData {
				var third, _ = strconv.Atoi(z)
				if first + second + third == 2020 {
					var result = first * second * third
					return result
				}
			}
		}
	}
	return 0
}

func readLines(inputFilename string) []string {
	file, err := os.Open(inputFilename)

	if err != nil {
		log.Printf("Failed to open file")
		log.Fatal(err)
	}
	scanner := bufio.NewScanner(file)
	scanner.Split(bufio.ScanLines)
	var text []string

	for scanner.Scan() {
		text = append(text, scanner.Text())
	}
	file.Close()
	return text
}